pootle/apps/virtualfolder/models.py

```diff
@@ -347,19 +275,156 @@ def get_adjusted_pootle_path(self, pootle_path):
         trail = pootle_path.replace(lead, '').lstrip('/')
         return '/'.join([lead, self.name, trail])
 
-    def get_translate_url(self, pootle_path, **kwargs):
-        """Get the translate URL for this virtual folder in the given path."""
-        adjusted_path = self.get_adjusted_pootle_path(pootle_path)
-        lang, proj, dp, fn = split_pootle_path(adjusted_path)
 
+class VFolderCachedTreeItemManager(models.Manager):
+    use_for_related_fields = True
+
+    def get_queryset(self):
+        return super(VFolderCachedTreeItemManager, self) \
+            .get_queryset().select_related('vfolder')
+
+
+class VFolderCachedTreeItem(models.Model, CachedTreeItem):
```

Comment:
I find `RootVirtualFolder` to be more confusing, since actually that is the virtual folder.

Also in the current stage there is no need for `CachedTreeItem` to be applied to that `RootVirtualFolder` model since we don't yet display whole stats for a virtual folder. But in case we do we still have to decide what `get_cachekey` is going to return for such items. Anyway this feels like it belongs to further developments to display virtual folders outside a specific TP.